packages/web-client/app/components/card-pay/deposit-workflow/transaction-status/index.ts

```diff
@@ -49,10 +58,25 @@ class CardPayDepositWorkflowTransactionStatusComponent extends Component<Workflo
         'completedLayer2TransactionReceipt',
         transactionReceipt
       );
-      this.completedCount = 3;
+      this.completedCount = 3; // steps 2 and 3 complete together
+      this.currentStep = this.completedCount;
+      this.waitForBlockConfirmations(transactionReceipt);
+    } catch (e) {
+      this.bridgeError = true;
+    }
+  }
+
+  async waitForBlockConfirmations(transactionReceipt: TransactionReceipt) {
```

Comment:
I know this isn't ready for review yet, but I just wanted to encourage you to convert this method to an ember-concurrency task, in order to handle component teardown well.